src/petals/client/inference_session.py

```diff
@@ -144,13 +139,24 @@ def step(
             if next_servers:
                 request_metadata["next_servers"] = next_servers
 
+        request_metadata["tensor_structure"] = tensor_structure
+
+        server_side_inference_schema, kwargs_schema = self.rpc_info["inference_schema"]
+        compression = server_side_inference_schema[0].compression
+        inference_schema = tuple(BatchTensorDescriptor.from_tensor(arg, compression) for arg in input_tensors)
```

Comment:
Can we use `nested_flatten()` here as the previous code does? This way, we'll support different compressions for each tensor and the code should become simpler

Reply:
Ok